File: Baekjoon/BFS/2606.java
// https://www.acmicpc.net/problem/2178

import java.util.*;
import java.io.*;

public class Main {
    static Scanner sc = new Scanner(System.in);
    static int n, m;
    static boolean[][] map;
    static boolean[] visited;
    static int count = 0;

    public static void main(String[] args) throws IOException {
        n = sc.nextInt();
        m = sc.nextInt();

        map = new boolean[n + 1][n + 1];
        visited = new boolean[n + 1];

        for(int i=0; i < m; i++) {
            int start = sc.nextInt();
            int end = sc.nextInt();
            map[start][end] = true;
            map[end][start] = true;
        }

        dfs(1);
        System.out.println(count);
    }

    public static void dfs(int start) {
        visited[start] = true;
        for(int i=1; i <= n; i++) {
            if(map[start][i] && !visited[i]) {
                count++;
                dfs(i);
            }
        }
    }
}

File: Baekjoon/BFS/2606.cpp
// https://www.acmicpc.net/problem/2606

#include <iostream>
#include <vector>
#include <queue>

using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    vector<vector<int>> adj(n + 1);
    vector<bool> visited(n + 1);
    for (int i = 0; i < m; i++) {
        int u, v;
        cin >> u >> v;
        adj[u].push_back(v);
        adj[v].push_back(u);
    }
    queue<int> q;
    q.push(1);
    visited[1] = true;
    int cnt = 0;
    while (!q.empty()) {
        int u = q.front();
        q.pop();
        cnt++;
        for (int v : adj[u]) {
            if (!visited[v]) {
                q.push(v);
                visited[v] = true;
            }
        }
    }
    cout << cnt - 1 << '\n';
    return 0;
}